compiletest: Require `//~` annotations even if `error-pattern` is specified

petrochenkov opened the pull request:
This is continuation of #138865 with some help from #139100.

`error-pattern` annotations that duplicate the newly added `//~` annotations are removed, other `error-pattern`s are not touched yet.

In exceptional cases `//@ compile-flags: --error-format=human` can be used to opt out of these checks.
In this PR I only had to use the opt out 3 times:
- `tests/ui/parser/utf16-{be,le}-without-bom.rs` - there are too many errors that are nearly identical (modulo location), because an error is reported on every second symbol
- `tests/ui-fulldeps/missing-rustc-driver-error.rs` - the errors list various rustc crate dependencies and may unexpectedly invalidate on random rustc changes

petrochenkov committed:
compiletest: Require `//~` annotations even if `error-pattern` is specified

jieyouxu commented:
r? jieyouxu

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14155738492/job/39654749452) [(plain)](https://github.com/rust-lang/rust/commit/73dd8ac1c22a18472c183b0a15cc45b27a4579bf/checks/39654749452/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04

ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
  make \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
#
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 2.907 Building wheels for collected packages: reuse
#12 2.908   Building wheel for reuse (pyproject.toml): started
#12 3.122   Building wheel for reuse (pyproject.toml): finished with status 'done'
#12 3.123   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132719 sha256=5bb60f62728aaedff7162745ce743c7f2f55069b3e7f82e6a37d70df455797cc
#12 3.123   Stored in directory: /tmp/pip-ephem-wheel-cache-4risttg6/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#12 3.126 Successfully built reuse
#12 3.126 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#12 3.529 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#12 3.530 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 4.074 Collecting virtualenv
#12 4.123   Downloading virtualenv-20.29.3-py3-none-any.whl (4.3 MB)
#12 4.280      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 27.8 MB/s eta 0:00:00
#12 4.343 Collecting filelock<4,>=3.12.2
#12 4.353   Downloading filelock-3.18.0-py3-none-any.whl (16 kB)
#12 4.375 Collecting distlib<1,>=0.3.7
#12 4.383   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.395      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 46.1 MB/s eta 0:00:00
#12 4.431 Collecting platformdirs<5,>=3.9.1
#12 4.438   Downloading platformdirs-4.3.7-py3-none-any.whl (18 kB)
#12 4.520 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.702 Successfully installed distlib-0.3.9 filelock-3.18.0 platformdirs-4.3.7 virtualenv-20.29.3
#12 4.703 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 DONE 4.8s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
DirectMap4k:      128960 kB
DirectMap2M:     8259584 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
[TIMING] core::build_steps::tool::LibcxxVersionTool { target: x86_64-unknown-linux-gnu } -- 0.218
---
fmt check
fmt: checked 5929 files
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/tests/ui/instrument-xray/flags-dupe-always.rs:8: trailing whitespace
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'venv'
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'virtualenv'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (25.0.1)
linting python files
All checks passed!
checking python file formatting
26 files already formatted
checking C++ file formatting
some tidy checks failed
Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:01:50
  local time: Sun Mar 30 12:00:43 UTC 2025
  network time: Sun, 30 Mar 2025 12:00:43 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

jieyouxu commented on src/doc/rustc-dev-guide/src/tests/ui.md:
```diff
@@ -335,6 +335,9 @@ But for strict testing, try to use the `ERROR` annotation as much as possible,
 including `//~?` annotations for diagnostics without span.
 For compile time diagnostics `error-pattern` should very rarely be necessary.
 
+Per-line annotations (`//~`) are still checked in tests using `error-pattern`,
+to opt out of these checks in exceptional cases use `//@ compile-flags: --error-format=human`.
```
Discussion: hm, that feels somewhat hacky (and difficult to infer from reading the test). Should we instead introduce sth more explicit like `//@ dont-check-per-line-annotations` (long, but its usage should be quite rare that this shouldn't be a problem)?

petrochenkov replied:
I think I'd rather use an existing option if it works, we already sort of have a proliferation of exotic options (#134888).

`--error-format=human` says "don't check the json, only the rendered text", which seems appropriate in cases like this. A new option is unlikely to be more discoverable.
(Perhaps we can reconsider when I migrate all of the remaining `error-pattern`s.)

jieyouxu replied:
Seems fair!

bors commented:
:umbrella: The latest upstream changes (presumably #139154) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

jieyouxu commented:
I'll look at this early tomorrow.

jieyouxu reviewed (approved):
Thanks, changes LGTM. Feel free to r=me after fixing the test conflict.

jieyouxu commented:
@rustbot author

bors commented:
:pushpin: Commit ecba3f8a838c409e4d75025db0dbc84adfe771d4 has been approved by `jieyouxu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=jieyouxu ecba3f8a838c409e4d75025db0dbc84adfe771d4 -->
<!-- homu: {"type":"Approved","sha":"ecba3f8a838c409e4d75025db0dbc84adfe771d4","approver":"jieyouxu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit ecba3f8a838c409e4d75025db0dbc84adfe771d4 with merge 9ddcc44be2b77ceefe452fb974a9c494e9e13737...
<!-- homu: {"type":"BuildStarted","head_sha":"ecba3f8a838c409e4d75025db0dbc84adfe771d4","merge_sha":"9ddcc44be2b77ceefe452fb974a9c494e9e13737"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/14220681559/job/39851764966)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/14220681559/job/39851764966","builder_name":"checks-actions"} -->

petrochenkov committed:
Fix up tests on wasm and msvc, and rebase conflicts

Can be fixed properly later by adding a new flag for non-exhaustive line annotation checking

bors commented:
:pushpin: Commit 5ca4dfcf541a89f1e65a4309ad7bdc6c93bccb21 has been approved by `jieyouxu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=jieyouxu 5ca4dfcf541a89f1e65a4309ad7bdc6c93bccb21 -->
<!-- homu: {"type":"Approved","sha":"5ca4dfcf541a89f1e65a4309ad7bdc6c93bccb21","approver":"jieyouxu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 5ca4dfcf541a89f1e65a4309ad7bdc6c93bccb21 with merge d8c54d43760a89ed18d1f02dbcf21c5dd9e52c21...
<!-- homu: {"type":"BuildStarted","head_sha":"5ca4dfcf541a89f1e65a4309ad7bdc6c93bccb21","merge_sha":"d8c54d43760a89ed18d1f02dbcf21c5dd9e52c21"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-19-3`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/14226355196/job/39866811028) [(plain)](https://github.com/rust-lang-ci/rust/commit/d8c54d43760a89ed18d1f02dbcf21c5dd9e52c21/checks/39866811028/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [ui] tests/ui/print-request/stability.rs#crate_root_lint_levels stdout ----

error in revision `crate_root_lint_levels`: /checkout/tests/ui/print-request/stability.rs:?: unexpected error: '?:?: ?:?: the `-Z unstable-options` flag must also be passed to enable the `crate-root-lint-levels` print option'

error in revision `crate_root_lint_levels`: 1 unexpected errors found, 0 expected errors not found
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/checkout/tests/ui/print-request/stability.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1" "--target=x86_64-unknown-linux-gnu" "--cfg" "crate_root_lint_levels" "--check-cfg" "cfg(test,FALSE,all_target_specs_json,crate_root_lint_levels,check_cfg,supported_crate_types,target_spec_json,calling_conventions,cfg,code_models,crate_name,deployment_target,file_names,host_tuple,link_args,native_static_libs,relocation_models,split_debuginfo,stack_protector_strategies,target_cpus,target_features,target_libdir,target_list,tls_models)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/print-request/stability.crate_root_lint_levels" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--print=crate-root-lint-levels"

--- unexpected errors (from JSON output) ---
ERROR     line   ?: ?:?: ?:?: the `-Z unstable-options` flag must also be passed to enable the `crate-root-lint-levels` print option
---

thread '[ui] tests/ui/print-request/stability.rs#crate_root_lint_levels' panicked at src/tools/compiletest/src/runtest.rs:799:13:
errors differ from expected
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: <compiletest::runtest::TestCx>::check_expected_errors
   3: <compiletest::runtest::TestCx>::run_revision
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/14226355196/job/39868942145)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/14226355196/job/39868942145","builder_name":"checks-actions"} -->

petrochenkov commented:
@bors r=jieyouxu

bors commented:
:pushpin: Commit 954d9444d57bf7701c7c25f7e1d8eb1433a3d3af has been approved by `jieyouxu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=jieyouxu 954d9444d57bf7701c7c25f7e1d8eb1433a3d3af -->
<!-- homu: {"type":"Approved","sha":"954d9444d57bf7701c7c25f7e1d8eb1433a3d3af","approver":"jieyouxu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 954d9444d57bf7701c7c25f7e1d8eb1433a3d3af with merge faac314f66f50d899d57c38ca4a82ea973a85067...
<!-- homu: {"type":"BuildStarted","head_sha":"954d9444d57bf7701c7c25f7e1d8eb1433a3d3af","merge_sha":"faac314f66f50d899d57c38ca4a82ea973a85067"} -->

rust-log-analyzer commented:
The job **`x86_64-msvc-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/14230233278/job/39879204981) [(plain)](https://github.com/rust-lang-ci/rust/commit/faac314f66f50d899d57c38ca4a82ea973a85067/checks/39879204981/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [ui] tests\ui\debuginfo\debuginfo-type-name-layout-ice-94961-2.rs stdout ----

error: D:/a/rust/rust/tests/ui/debuginfo/debuginfo-type-name-layout-ice-94961-2.rs:?: unexpected error: '?:?: ?:?: values of the type `[u8; usize::MAX]` are too big for the target architecture'

error: D:/a/rust/rust/tests/ui/debuginfo/debuginfo-type-name-layout-ice-94961-2.rs:1: expected error not found: values of the type `[u8; usize::MAX]` are too big for the target architecture

error: 1 unexpected errors found, 1 expected errors not found
status: exit code: 1
command: PATH="D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin;C:\Program Files (x86)\Windows Kits\10\bin\x64;C:\Program Files (x86)\Windows Kits\10\bin\10.0.26100.0\x64;C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\bin\HostX64\x64;C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\bin\HostX64\x64;D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0-bootstrap-tools\x86_64-pc-windows-msvc\release\deps;D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0\bin;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Users\runneradmin\bin;D:\a\rust\rust\ninja;D:\a\rust\rust\sccache;C:\Program Files\MongoDB\Server\7.0\bin;C:\vcpkg;C:\tools\zstd;C:\hostedtoolcache\windows\stack\3.5.1\x64;C:\cabal\bin;C:\ghcup\bin;C:\mingw64\bin;C:\Program Files\dotnet;C:\Program Files\MySQL\MySQL Server 8.0\bin;C:\Program Files\R\R-4.4.2\bin\x64;C:\SeleniumWebDrivers\GeckoDriver;C:\SeleniumWebDrivers\EdgeDriver;C:\SeleniumWebDrivers\ChromeDriver;C:\Program Files (x86)\sbt\bin;C:\Program Files (x86)\GitHub CLI;C:\Program Files\Git\usr\bin;C:\Program Files (x86)\pipx_bin;C:\npm\prefix;C:\hostedtoolcache\windows\go\1.23.7\x64\bin;C:\hostedtoolcache\windows\Python\3.9.13\x64\Scripts;C:\hostedtoolcache\windows\Python\3.9.13\x64;C:\hostedtoolcache\windows\Ruby\3.3.7\x64\bin;C:\Program Files\OpenSSL\bin;C:\tools\kotlinc\bin;C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.14-7\x64\bin;C:\Program Files\ImageMagick-7.1.1-Q16-HDRI;C:\Program Files\Microsoft SDKs\Azure\CLI2\wbin;C:\ProgramData\kind;C:\ProgramData\Chocolatey\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\Program Files\PowerShell\7;C:\Program Files\Microsoft\Web Platform Installer;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn;C:\Program Files\Microsoft SQL Server\150\Tools\Binn;C:\Program Files\dotnet;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit;C:\Program Files (x86)\WiX Toolset v3.14\bin;C:\Program Files\Microsoft SQL Server\130\DTS\Binn;C:\Program Files\Microsoft SQL Server\140\DTS\Binn;C:\Program Files\Microsoft SQL Server\150\DTS\Binn;C:\Program Files\Microsoft SQL Server\160\DTS\Binn;C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin;C:\Program Files\CMake\bin;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.9.9\bin;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\nodejs;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Program Files\GitHub CLI;C:\tools\php;C:\Program Files (x86)\sbt\bin;C:\Program Files\Amazon\AWSCLIV2;C:\Program Files\Amazon\SessionManagerPlugin\bin;C:\Program Files\Amazon\AWSSAMCLI\bin;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files\mongosh;C:\Program Files\LLVM\bin;C:\Program Files (x86)\LLVM\bin;C:\Users\runneradmin\.dotnet\tools;C:\Users\runneradmin\.cargo\bin;C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\bin\HostX64\x64" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage2\\bin\\rustc.exe" "D:\\a\\rust\\rust\\tests\\ui\\debuginfo\\debuginfo-type-name-layout-ice-94961-2.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=C:\\Users\\runneradmin\\.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=D:\\a\\rust\\rust\\vendor" "--sysroot" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage2" "--target=x86_64-pc-windows-msvc" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "--out-dir" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\test\\ui\\debuginfo\\debuginfo-type-name-layout-ice-94961-2" "-A" "unused" "-A" "internal_features" "-Crpath" "-Lnative=D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\native\\rust-test-helpers" "-C" "debuginfo=2"

--- unexpected errors (from JSON output) ---
ERROR     line   ?: ?:?: ?:?: values of the type `[u8; usize::MAX]` are too big for the target architecture
---
--- not found errors (from test file) ---
ERROR     line   1: values of the type `[u8; usize::MAX]` are too big for the target architecture
---


thread '[ui] tests\ui\debuginfo\debuginfo-type-name-layout-ice-94961-2.rs' panicked at src\tools\compiletest\src\runtest.rs:799:13:
errors differ from expected
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- [ui] tests\ui\panic-runtime\unwind-tables-target-required.rs stdout ----

error: D:/a/rust/rust/tests/ui/panic-runtime/unwind-tables-target-required.rs:?: unexpected error: '?:?: ?:?: target requires unwind tables, they cannot be disabled with `-C force-unwind-tables=no`'

error: 1 unexpected errors found, 0 expected errors not found
status: exit code: 1
command: PATH="D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin;C:\Program Files (x86)\Windows Kits\10\bin\x64;C:\Program Files (x86)\Windows Kits\10\bin\10.0.26100.0\x64;C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\bin\HostX64\x64;C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\bin\HostX64\x64;D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0-bootstrap-tools\x86_64-pc-windows-msvc\release\deps;D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0\bin;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Users\runneradmin\bin;D:\a\rust\rust\ninja;D:\a\rust\rust\sccache;C:\Program Files\MongoDB\Server\7.0\bin;C:\vcpkg;C:\tools\zstd;C:\hostedtoolcache\windows\stack\3.5.1\x64;C:\cabal\bin;C:\ghcup\bin;C:\mingw64\bin;C:\Program Files\dotnet;C:\Program Files\MySQL\MySQL Server 8.0\bin;C:\Program Files\R\R-4.4.2\bin\x64;C:\SeleniumWebDrivers\GeckoDriver;C:\SeleniumWebDrivers\EdgeDriver;C:\SeleniumWebDrivers\ChromeDriver;C:\Program Files (x86)\sbt\bin;C:\Program Files (x86)\GitHub CLI;C:\Program Files\Git\usr\bin;C:\Program Files (x86)\pipx_bin;C:\npm\prefix;C:\hostedtoolcache\windows\go\1.23.7\x64\bin;C:\hostedtoolcache\windows\Python\3.9.13\x64\Scripts;C:\hostedtoolcache\windows\Python\3.9.13\x64;C:\hostedtoolcache\windows\Ruby\3.3.7\x64\bin;C:\Program Files\OpenSSL\bin;C:\tools\kotlinc\bin;C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.14-7\x64\bin;C:\Program Files\ImageMagick-7.1.1-Q16-HDRI;C:\Program Files\Microsoft SDKs\Azure\CLI2\wbin;C:\ProgramData\kind;C:\ProgramData\Chocolatey\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\Program Files\PowerShell\7;C:\Program Files\Microsoft\Web Platform Installer;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn;C:\Program Files\Microsoft SQL Server\150\Tools\Binn;C:\Program Files\dotnet;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit;C:\Program Files (x86)\WiX Toolset v3.14\bin;C:\Program Files\Microsoft SQL Server\130\DTS\Binn;C:\Program Files\Microsoft SQL Server\140\DTS\Binn;C:\Program Files\Microsoft SQL Server\150\DTS\Binn;C:\Program Files\Microsoft SQL Server\160\DTS\Binn;C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin;C:\Program Files\CMake\bin;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.9.9\bin;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\nodejs;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Program Files\GitHub CLI;C:\tools\php;C:\Program Files (x86)\sbt\bin;C:\Program Files\Amazon\AWSCLIV2;C:\Program Files\Amazon\SessionManagerPlugin\bin;C:\Program Files\Amazon\AWSSAMCLI\bin;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files\mongosh;C:\Program Files\LLVM\bin;C:\Program Files (x86)\LLVM\bin;C:\Users\runneradmin\.dotnet\tools;C:\Users\runneradmin\.cargo\bin;C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\bin\HostX64\x64" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage2\\bin\\rustc.exe" "D:\\a\\rust\\rust\\tests\\ui\\panic-runtime\\unwind-tables-target-required.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=C:\\Users\\runneradmin\\.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=D:\\a\\rust\\rust\\vendor" "--sysroot" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage2" "--target=x86_64-pc-windows-msvc" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\test\\ui\\panic-runtime\\unwind-tables-target-required" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\native\\rust-test-helpers" "-C" "force-unwind-tables=no"

--- unexpected errors (from JSON output) ---
ERROR     line   ?: ?:?: ?:?: target requires unwind tables, they cannot be disabled with `-C force-unwind-tables=no`
---

thread '[ui] tests\ui\panic-runtime\unwind-tables-target-required.rs' panicked at src\tools\compiletest\src\runtest.rs:799:13:
errors differ from expected

---
test result: FAILED. 18513 passed; 2 failed; 282 ignored; 0 measured; 23 filtered out; finished in 1105.59s

Some tests failed in compiletest suite=ui mode=ui host=x86_64-pc-windows-msvc target=x86_64-pc-windows-msvc
Build completed unsuccessfully in 1:35:15
make: *** [Makefile:113: ci-msvc-py] Error 1
  local time: Wed Apr  2 23:30:43 CUT 2025
  network time: Wed, 02 Apr 2025 23:30:43 GMT
##[error]Process completed with exit code 2.
Post job cleanup.
[command]"C:\Program Files\Git\bin\git.exe" version
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/14230233278/job/39883237695)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/14230233278/job/39883237695","builder_name":"checks-actions"} -->

petrochenkov commented:
@bors r=jieyouxu

bors commented:
:hourglass: Testing commit 4916d44b59a2b5347506b4551b389fe902bdf54b with merge 946aea0b3d9f43e57953ce5094d8e4a2a244ce71...
<!-- homu: {"type":"BuildStarted","head_sha":"4916d44b59a2b5347506b4551b389fe902bdf54b","merge_sha":"946aea0b3d9f43e57953ce5094d8e4a2a244ce71"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/14242509076/job/39929955645)
Approved by: jieyouxu
Pushing 946aea0b3d9f43e57953ce5094d8e4a2a244ce71 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"jieyouxu","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/14242509076/job/39929955645"},"merge_sha":"946aea0b3d9f43e57953ce5094d8e4a2a244ce71"} -->

rust-timer commented:
Finished benchmarking commit (946aea0b3d9f43e57953ce5094d8e4a2a244ce71): [comparison URL](https://perf.rust-lang.org/compare.html?start=e0883a2a6c8e4afcb6e26a182885f687ba63a7f5&end=946aea0b3d9f43e57953ce5094d8e4a2a244ce71&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=e0883a2a6c8e4afcb6e26a182885f687ba63a7f5&end=946aea0b3d9f43e57953ce5094d8e4a2a244ce71&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=e0883a2a6c8e4afcb6e26a182885f687ba63a7f5&end=946aea0b3d9f43e57953ce5094d8e4a2a244ce71&stat=max-rss)
<details>
<summary>Results (primary -2.5%, secondary -2.1%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -2.5% | [-2.5%, -2.5%] | 1     |
| Improvements ✅ <br /> (secondary) | -2.1% | [-2.5%, -1.8%] | 2     |
| All ❌✅ (primary)                 | -2.5% | [-2.5%, -2.5%] | 1     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=e0883a2a6c8e4afcb6e26a182885f687ba63a7f5&end=946aea0b3d9f43e57953ce5094d8e4a2a244ce71&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.

### [Binary size](https://perf.rust-lang.org/compare.html?start=e0883a2a6c8e4afcb6e26a182885f687ba63a7f5&end=946aea0b3d9f43e57953ce5094d8e4a2a244ce71&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 777.235s -> 776.226s (-0.13%)
**Artifact size**: 365.89 MiB -> 365.90 MiB (0.00%)
